### PR TITLE
chore: remove java owlbot

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Rmove Java Owlbot since the source code is migrated to sdk-platform-java and it is disabled in non-GAPIC libraries (see go/disable-owlbot-java).

### CODE REVIEW

File: tests/test_language_java.py
Comment (blakeli0) [inline tests/test_language_java.py:1]: I don't see this test being migrated to sdk-platform-java and looks like it still provides value, can you try to move it?

File: docker/owlbot/java/cloudbuild.yaml
Comment (blakeli0) [inline docker/owlbot/java/cloudbuild.yaml:1]: I assume there are some cloud build triggers associated with these cloud build yamls? Can we figure out what are them and delete them as well?

Comment (JoeWang1127) [reply docker/owlbot/java/cloudbuild.yaml:1]: I created b/368584265 so @chingor13 can remove it.

File: tests/test_language_java.py
Comment (JoeWang1127) [reply tests/test_language_java.py:1]: Traced in https://github.com/googleapis/sdk-platform-java/pull/3227

**Review by blakeli0 — APPROVED** [review]

**Review by chingor13 — APPROVED** [review]

**Pull request merged**